hawk/src/main/java/com/orhanobut/hawk/SqliteStorage.java

```diff
@@ -92,10 +93,15 @@ public SqliteHelper(Context context, String dbName) {
 
     public synchronized boolean put(String key, String value) {
       SQLiteDatabase db = this.getWritableDatabase();
-      db.execSQL("INSERT OR REPLACE INTO " + TABLE_NAME +
-          " (" + COL_KEY + ", " + COL_VALUE + ") " +
-          " VALUES('" + key + "', '" + value + "')");
-      db.close();
+      try {
+        db.execSQL("INSERT OR REPLACE INTO " + TABLE_NAME +
+                " (" + COL_KEY + ", " + COL_VALUE + ") " +
+                " VALUES('" + key + "', '" + value + "')");
+      } catch (SQLException ignored) {
+        return false;
```

Comment:
in this case, db won't be closed, we can keep a local variable for the result. Also maybe a wider exception would be better idea here if anything rather than sql exception happens. Since we don't deal with the exception itself.